### ISSUE TITLE
unexpected "break" behavior in static for loop

### ISSUE THREAD

**Issue by RitChan** [taichi-dev/taichi#5982]
**Describe the bug**
Taichi kernel executed an impossibile "break" statement inside static for loop

**To Reproduce**
```py
import taichi as ti


@ti.kernel
def main():
    if 1 > 0:
        for i in ti.static(range(3)):
            print("Taichi 1")
            if 1 < 0:
                break
            print("Taichi 2")


if __name__ == '__main__':
    ti.init(ti.cpu)
    main()
```

**Log/Screenshots**

```
$ python my_sample_code.py
[Taichi] version 1.1.2, llvm 10.0.0, commit f25cf4a2, win, python 3.7.9
[Taichi] Starting on arch=x64
Taichi 1
```

**Additional comments**
Removing `ti.static` gives the correct result:

- Code:
```py
import taichi as ti


@ti.kernel
def main():
    if 1 > 0:
        for i in range(3):
            print("Taichi 1")
            if 1 < 0:
                break
            print("Taichi 2")


if __name__ == '__main__':
    ti.init(ti.cpu)
    main()
```

- Output:
```py
[Taichi] version 1.1.2, llvm 10.0.0, commit f25cf4a2, win, python 3.7.9
[Taichi] Starting on arch=x64
Taichi 1
Taichi 2
Taichi 1
Taichi 2
Taichi 1
Taichi 2
```

**Comment by lucifer1004** [comment]
Can you upgrade to the latest version (1.1.2) and check whether the bug still exists?

**Comment by RitChan** [comment]
> Can you upgrade to the latest version (1.1.2) and check whether the bug still exists?

I did it. And I have updated issue descriptions above.

My temporary workround is to unroll static loop manually😂

**Comment by lucifer1004** [comment]
The initial IR is incorrect.

```python
@ti.kernel
def main():
    for i in ti.static(range(3)):
        print("Taichi 1")
        if 1 < 0:
            break
        print("Taichi 2")
```

yields

```assembly
[I 09/09/22 13:18:49.703 2894195] [compile_to_offloads.cpp:operator()@22] [main_c64_0] Initial IR:
kernel {
  print "Taichi 1\n"
  $1 : if 0 {
  } else {
  }
}
```

**Comment by lucifer1004** [comment]
`continue` can also cause issues. For example,

```python
@ti.kernel
def main():
    for i in ti.static(range(3)):
        print("Taichi 1")
        if False:
            continue
        else:
            print("Taichi 2")
```

yields

```assembly
[I 09/09/22 13:20:40.614 2896012] [compile_to_offloads.cpp:operator()@22] [main_c64_0] Initial IR:
kernel {
  print "Taichi 1\n"
  $1 : if 0 {
  } else {
  }
  print "Taichi 1\n"
  $3 : if 0 {
  } else {
  }
  print "Taichi 1\n"
  $5 : if 0 {
  } else {
  }
}
```

**Comment by lucifer1004** [comment]
I've found some clues. It seems that to make the code above work, we need to also wrap the conditions in `ti.static`.

**Comment by lucifer1004** [comment]
For example,

```python
@ti.kernel
def main():
    for i in ti.static(range(3)):
        print("Taichi 1")
        if ti.static(1 < 0):
            break
        print("Taichi 2")
```

works as expected.

**Comment by lucifer1004** [comment]
Related source code: https://github.com/taichi-dev/taichi/blob/c32970fc634c74617acd8e2c017fef37f42988c9/python/taichi/lang/ast/ast_transformer.py#L1325

**Comment by lucifer1004** [comment]
Seems that this issue is a duplicate of #5729 and has been fixed by #5755

**Comment by RitChan** [comment]
That's good. So dynamic "if" will not be supported in static "for" in future version. And for such case, user will have to manually unroll the loop to allow dynamic "if" to work properly.
